src/addmult.c

```diff
@@ -336,57 +330,63 @@ void add_mult_line(char *line) {
 int init_and_load_multipliers(void) {
 
     FILE *cfp;
-    char s_inputbuffer[2000] = "";
+    char *s_inputbuffer = NULL;
+    size_t s_inputbuffer_len = 0;
     char *mults_location;
+    ssize_t read;
 
     if (mults_possible) {
-	/* free old array if exists */
-	g_ptr_array_free(mults_possible, TRUE);
+        /* free old array if exists */
+        g_ptr_array_free(mults_possible, TRUE);
     }
     mults_possible = g_ptr_array_new_with_free_func(free_possible_mult);
 
-
     if (strlen(multsfile) == 0) {
-	return 0;
+	    return 0;
     }
 
     mults_location = find_available(multsfile);
 
     if ((cfp = fopen(mults_location, "r")) == NULL) {
-	mvprintw(9, 0, "Error opening multiplier file %s.\n", multsfile);
-	refreshp();
-	sleep(5);
+        mvprintw(9, 0, "Error opening multiplier file %s.\n", multsfile);
+        refreshp();
+        sleep(5);
     }
 
     g_free(mults_location);
 
     if (cfp == NULL) {
-	return 0;       // couldn't open file
+	    return 0;       // couldn't open file
     }
 
-    while (fgets(s_inputbuffer, sizeof(s_inputbuffer), cfp) != NULL) {
-
-	/* strip leading and trailing whitespace */
-	g_strstrip(s_inputbuffer);
-
-	/* drop comments starting with '#' and empty lines */
-	if (*s_inputbuffer == '#' || *s_inputbuffer == '\0') {
-	    continue;
-	}
-
-	add_mult_line(s_inputbuffer);
-
+    while ((read = getline(&s_inputbuffer, &s_inputbuffer_len, cfp)) != -1) {
+            if (s_inputbuffer_len > 0) {
```

Comment:
Instead of moving the good case to an extra indented block
```
if (all_is_ok==true) {
  // the good case
   step1();
   step2();
   ...
} else {
  do_error_handling();
}
```

consider  bailing-out early and keep the good case minimally indented:
```
if (all_is_ok == false) {
   do_error_handling();
   break;  // or return or ...
}

// the good case
step1();
step2();
```